server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -140,6 +169,16 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetadata.getNumberOfShards() * (1 + indexMetadata.getNumberOfReplicas());
     }
 
+    /**
+     * Returns true if the cluster.ignore_hidden_indexes property is set to true and the index name
+     * starts with '.' else false.
+     *
+     * @param indexName  The index which needs to be validated.
+     */
+    private boolean validateHiddenIndex(String indexName) {
+        return this.ignoreHiddenIndexes && indexName.charAt(0) == '.';
```

Comment:
`indexName.charAt(0) == '.'`  could this be little more generic where we could have pattern match or list of pattern match to allow user customisations?

Reply:
This will only be done for hidden and system indexes and index names starting with a dot should be reserved for hidden  and system indices

Reply:
should we exclude datastream indices from this ? we can use `BACKING_INDEX_PREFIX`  to match prefix for it .

Reply:
Ack will add that

Reply:
hidden should be index.hidden in index metadata